libpod/container_config.go

```diff
@@ -400,4 +403,7 @@ type ContainerMiscConfig struct {
 	// InitContainerType specifies if the container is an initcontainer
 	// and if so, what type: always or once are possible non-nil entries
 	InitContainerType string `json:"init_container_type,omitempty"`
+	// FinalSpecGen is the configuration the container used to generate the spec as well as the
+	// config. This is used to pass options down from infra container to other containers joing a pod
+	FinalSpecGen *specgen.SpecGenerator `json:"final_spec_gen,omitempty"`
```

Comment:
I do not like to store the full specgen in a container config, @mheon WDYT?

Reply:
Yeah, this seems excessive

Reply:
Honestly this seems like it should be handled in Specgen itself, not in Libpod. We can store the pod's security opts as labels (we're doing that for some things in container inspect right now) and retrieve them as part of spec generation, adding the container to the pod.

Reply:
yeah I am going to convert this to a draft. I want to address the inheritance of multiple infra config items in a more elegant way. I have kind of been taking this approach with a `...From` array that funnels it to the container but entities like this complicate that a bit.

Reply:
@mheon had an idea to add an intermediate struct called something like `InfraInherit` that contains the currently supported pod create config entities. This could then be used in `container_create.go` to do an marshal/unmarhal dance into our new container config. This could replace all of the `VolumesFrom` and `DevicesFrom` entities and streamline it into a single function. Would this make sense?